src/hooks/useFlipCard.ts

```diff
@@ -0,0 +1,24 @@
+import React, { useEffect, useState } from 'react';
+
+interface UseFlipCardProps {
+  frontDeps: Array<React.ComponentState>;
+  backDeps: Array<React.ComponentState>;
+}
+
+export default function useFlipCard({ frontDeps, backDeps }: UseFlipCardProps) {
+  const [isFrontSide, setFrontSide] = useState(true);
+
+  const flipCard = () => {
+    setFrontSide((prev) => !prev);
+  };
+
+  useEffect(() => {
+    setFrontSide(false);
+  }, [JSON.stringify(backDeps)]);
```

Comment:
### 질문 2
카드가 뒤집힌 상태를 UI 구현과 분리하기 위해 해당 훅을 만들었습니다.

데이터가 변경되었을 때 카드의 앞면이 보여야 하는 변수들을 frontDeps, 뒷면이 보여야 하는 변수들을 backDeps로 전달받고 있는데요,
useEffect의 의존성 배열로 그대로 넣어주면 props 중 하나만 바뀌어도 새로운 객체가 만들어지는 props 객체의 특성상 의도한대로 작동하지 않았습니다.

따라서 약간의 꼼수(?)로 JSON.stringfy로 처리해주었는데, 이렇게 작성하는 것이 괜찮은지 혹은 더 좋은 방법이 있을지 궁금해 하루의 의견이 궁금합니다!

Reply:
저도 배열에서 변경된 값이 있는지 여부로 effect를 호출하기 위해서 `[JSON.stringify(array)]` 형태로 작성한 적이 있어요. 렛서가 넣어주신 deps에도 모두 array들로 구성되어 있어서 의도한 대로 동작할 것 같아요. 👍

Reply:
렛서가 작성해주신 것과 기능상의 차이는 있지만, CVC 작성을 마치면 바로 앞면으로 돌아오도록 하는 플로우도 괜찮다면 이렇게 구현할 수도 있을 것 같아요.

- CVC Input focus 이벤트 발생 시 -> `setBackSide(true)`
- CVC Input blur 이벤트 발생 시 -> `setBackSide(false)`

<br/>

effect 관련 좋은 자료가 생각나 같이 첨부해보아요.
언제 effect를 사용할지, 언제 event handling만으로 충분할지 정리해보기 좋은 글이에요 :)

- ⭐️추천⭐️: https://react.dev/learn/you-might-not-need-an-effect

> When you’re not sure whether some code should be in an Effect or in an event handler, ask yourself why this code needs to run. Use Effects only for code that should run because the component was displayed to the user. In this example, the notification should appear because the user pressed the button, not because the page was displayed! Delete the Effect and put the shared logic into a function called from both event handlers